src/com/projectkorra/projectkorra/ability/ElementalAbility.java

```diff
@@ -45,8 +59,34 @@ public boolean isTransparent(final Block block) {
 		return isTransparent(this.player, this.getName(), block);
 	}
 
-	public List<String> getEarthbendableBlocks() {
-		return getConfig().getStringList("Properties.Earth.EarthBlocks");
+	public static void clearBendableMaterials() {
+		EARTH_BLOCKS.clear();
+		ICE_BLOCKS.clear();
+		METAL_BLOCKS.clear();
+		PLANT_BLOCKS.clear();
+		SAND_BLOCKS.clear();
+		SNOW_BLOCKS.clear();
+	}
+
+	public static List<String> getEarthbendableBlocks() {
+		return new ArrayList<String>(EARTH_BLOCKS);
+	}
+
+	public static void addTags(Set<String> outputSet, List<String> configList) {
+		ListIterator<String> iterator = new ArrayList<String>(configList).listIterator();
+		iterator.forEachRemaining(next -> {
+			if (next.startsWith("#")) {
+				NamespacedKey key = NamespacedKey.minecraft(next.replaceFirst("#", ""));
+				if (!Bukkit.getTag(Tag.REGISTRY_BLOCKS, key, Material.class).getValues().isEmpty()) {
+					for (Material material : Bukkit.getTag(Tag.REGISTRY_BLOCKS, key, Material.class).getValues()) {
+						outputSet.add(material.toString());
+					}
+				}
+			} else {
+				outputSet.add(next.toUpperCase());
+			}
+		});
+		configList.clear();
```

Comment:
What was the reasoning behind clearing the config list?

Reply:
Ok, looked at http://wiki.c2.com/?MarkAndSweep to make sure. It's not necessary to clear the list in this case, I understand now. But is it necessary to question a good habit? lol  The reasoning is that I do not know for certain getConfig().getStringList() remains referenced in the memory or not, so I figured it couldn't hurt to just clear it in case. I'm not trying to outsmart GC here. lol

Reply:
Took a look at MemorySection in spigot to be double positive. It's safe to just not clear. But seriously, typing configList.clear() saved me so much more time than doing all of that. Call it a no-op or not, it's just minute details.